### PR TITLE
New SAMPLing files

### PR DESCRIPTION
**Pull request by andrrizzi**
Here are the files for the SAMPLing challenge! I need to add a couple more things to the README, but the files should be ready.

Few things to notice:
1. The 10mM experimental ionic strength for OA was too low to add any ion, so only the Na+ necessary for neutralization are added to the OA-guest systems.
2. Could you double check the atom map in `host_guest/SAMPLing/atom_map.json` and see if it's reasonable?
3. I tweaked OpenEye to give me reasonably different docked poses, and each pose has gone through 1ns of equilibration so the initial configurations of the 5 replicates should be fairly different from each other.

### CODE REVIEW

**Comment by mrshirts** [comment]
Hmm. This looks like I'll need to reconvert everything?

**Comment by davidlmobley** [comment]
@mrshirts - did you deposit conversion utilities? I'm thinking the easiest (and best) path forward here is to (a) use what you already did in terms of energy validations, etc., so that doesn't have to be repeated, but (b) re-do conversion using Andrea's updated files. Basically, Andrea and I realized last week sometime that he wanted to make sure the files in the repo correspond to what he's actually going to run for his reference calculations, so he wanted to make a round of updates. I believe it's mainly changes to protonation state/system composition (no changes to force field or anything similar) so it shouldn't have functional differences that are likely to affect the correctness of conversion.

To put it another way, one approach to minimize the extra work required on your end would be to:
1) Get your PR merged in
2) Do a new point release of the data for you to reference in your energy evaluation content, saying that you did the energy validation on that particular version
3) Merge Andrea's content in and then convert again and do another new point release

That way the energy validation (which I'm assuming is the most time consuming part?) would be on a specific, useful point release and we'd be able to proceed without you having to backtrack too much.

**Comment by mrshirts** [comment]
Energy validation only takes a few hours for everyhthing, and I've got all
the kinks worked out, so it's push and run at this point.

I didn't deposit the conversion utilities because to actually do the energy
checks, you have to have all of your programs in the right place.

If we wanted to set it up so people could convert themselves, we could do
that.  But I don't know we need to.  It's more useful to have the
conversions pegged to the energy.

The most time consuming part is going to be prepping the data so it's what
we want end users to see.  If you can take a look at what's in the pull
request (which is basically all the output files (including things like the
.xvg and .log files of the energy validation and the direct InterMol
output), you can let me know how it should be reorganized (and how to
divide up that reorganization).

**Comment by davidlmobley** [comment]
Thanks, @mrshirts , I have time blocked out to do this around 2 or 2:30 this afternoon. Sorry for the duplication of effort you're forced to here -- it's totally my fault. I had in my head the files were final, but then when Andrea wanted to re-generate them to match his final protocol I totally forgot about the fact that you were working on conversions simultaneously.

**Comment by andrrizzi** [comment]
So sorry for the lack of coordination, @mrshirts . I had to make changes to the files following the consensus in #4 and #9 to provide 5 different initial configurations for each host-guest and to keep a constant number of waters in OA systems to allow relative free energy calculations.

The rest of the pipeline is almost unchanged. Let me know if there's anything I can help with.

This is ready for review on my end.

**Comment by davidlmobley** [comment]
@andrrizzi - maybe I'm missing something, but on this point:

> Could you double check the atom map in host_guest/SAMPLing/atom_map.json and see if it's reasonable?

It looks to me like you've mapped these backwards. Specifically, `unique_atoms_G6` contains `[1, 2, 5, ...` where, at least from what I can tell in my viewer, G6 atom `1` is part of the carboxylic acid. But this is the single most important group in determining the guest orientation/position, most likely, so you would want to do the mapping in a way which preserves the carboxylic acid and mutates other atoms which are more distant.

I'm trying to see what you've mapped. The carboxylic acid in G3 is atoms `4`, `1` and `3`; from the atom map, this ends up being mapped to G6 atoms `3`, None, and None. G6 atom 3 is a proton in the carboxylic acid, though, so....??? 

(Note for mapping purposes I am reading your json file by hand and comparing to a visualization of the PDB.)

So it seems to me that one of these must be going on
a) I am doing something wrong in interpreting your mapping
b) You have mapped incorrectly
c) The atom ordering is getting shuffled in some of your files

**Comment by jmichel80** [comment]
Hi guys,

Picking up on a comment from @steboss

''The 10mM experimental ionic strength for OA was too low to add any ion, so only the Na+ necessary for neutralization are added to the OA-guest systems.''

--> Yes but the experimental buffer was 10 mM Na3PO4 so to match that ionic strength with NaCl you need 60 mM.  This results in 4 Na  and Cl  to add to OA in @steboss box setup.

**Comment by davidlmobley** [comment]
The equilibrated structures look good. Comments:
- CB8-G3: Three of your five binding modes are pretty similar, but one switches to point out the other face and one is pretty different. So they look good to me.
- OA-G3: Looks good. Not a lot of diversity, but G3 is pretty simple.
- OA-G6: Great, more diversity here. 

Did you want to add README material relating to SAMPLing (perhaps in the main README.md, linking to the content in SAMPLing) which would explain the SAMPLing challenge and how we intend it to be done, @andrrizzi (recapitulating our GitHub discussion)? Then this could be a draft of the content I would also e-mail out, or I could adapt for e-mailing out.

The main README.md does need updating to reflect the SAMPLing challenge, though let me know if you'd prefer that I do that.

We'll also need to start thinking of a submission format, though we can do that after getting this material in place.

**Comment by davidlmobley** [comment]
So, overall, @andrrizzi , looks like we need to deal with the atom mapping, and you or I needs to update some documentation (especially on the main page to reflect what the SAMPLing challenge IS!) and then we can get this merged.

**Comment by mrshirts** [comment]
For what it's worth for host-guest systems: our main insight from tests of
this (which I'm sure other people have) is that the slowest degree of
freedom is the emptying of the host of water for the guest to bind.
Monitoring that will give you the best sense of convergence.

We've been using expanded ensemble to try to  speed this up, though it
doesn't work perfectly.

**Comment by davidlmobley** [comment]
One other thought, though, @andrrizzi - did you WANT to go ahead and also retain/deposit here the input files for reference calculations you will be running? One might imagine that some people will not be interested in exploring variations of system prep and would be happy to run the same things you are running even if not targeting the SAMPLing challenge. Previously you had those inputs here, but i believe this PR would remove them?

(If indeed we remove them, then that dramatically reduces the scale of the conversion effort in #12 )

**Comment by andrrizzi** [comment]
> Specifically, unique_atoms_G6 contains [1, 2, 5, ... where, at least from what I can tell in my viewer, G6 atom 1 is part of the carboxylic acid

The atoms indices are 0-based, so probably atom 1 in your viewer will be atom 2. I specified it in the README, but would people maybe prefer these to be 1-based?

> --> Yes but the experimental buffer was 10 mM Na3PO4 so to match that ionic strength with NaCl you need 60 mM. This results in 4 Na and Cl to add to OA in @Steboss box setup.

Ah! Sorry, I missed this. I can regenerate the files with this ionic strength if you want. It would take me only a couple of hours.

> The main README.md does need updating to reflect the SAMPLing challenge, though let me know if you'd prefer that I do that.

It's the same for me, go ahead if you prefer to do it, otherwise, I can work on the first draft of this tonight, no problem. Feel free to make any edit in this PR.

> did you WANT to go ahead and also retain/deposit here the input files for reference calculations you will be running?

Which reference calculations are you referring to? My plan was to run the SAMPLing reference calculations from the files that I have uploaded here. Or do you mean the YAML scripts I'll use for YANK?

**Comment by davidlmobley** [comment]
@andrrizzi - my comments above were AFTER accounting for zero based indexing. That is, I STILL think the mappings don't make sense to me.

> Ah! Sorry, I missed this. I can regenerate the files with this ionic strength if you want. It would take me only a couple of hours.

Yes, we want to try and match that ionic strength.

>> did you WANT to go ahead and also retain/deposit here the input files for reference calculations you will be running?

>Which reference calculations are you referring to? My plan was to run the SAMPLing reference calculations from the files that I have uploaded here. Or do you mean the YAML scripts I'll use for YANK?

To be clear, I thought we were planning that you would ultimately:
a) Run a set of reference calculations on all of the compounds in the SAMPL set, which would benchmark performance/accuracy of Yank and provide an overall point of comparison for all of the participants; this would serve for the full challenge, not the SAMPLing component specifically. 
b) Provide reference values for the SAMPLing component of the challenge (where I thought you'd said you wanted to use the same files you were going to be using for your reference calculations of (a)). 

If you want to discuss briefly, ping me on Slack and we can have a quick call.

**Comment by davidlmobley** [comment]
OK, so Andrea and I have sorted out the mapping issue, which was just a problem with my viewer. Currently the plan is to:
- Correct ionic strength in this PR
- Slightly update atom mapping to better handle stereochemistry
- Get updated files for reference calculations (beyond the SAMPLing challenge ones) in a separate PR

**Comment by mrshirts** [comment]
OK, let me know when you when things are finalized.  I am happy to convert everything at this point; it's all going through smoothly.

**Comment by andrrizzi** [comment]
@davidlmobley I've added a draft of the SAMPLing challenge description to the main README. Feel free to edit as needed.

**Comment by davidlmobley** [comment]
@andrrizzi - I'll make some minor edits to this now, but what about the issue with ionic strength? Can you resolve it so we can get this going??

**Comment by andrrizzi** [comment]
Whops! Sorry, I had the local commits but somehow I forgot to push them (and I modified the README directly on github). They should be up now.

**Comment by davidlmobley** [comment]
Thanks, @andrrizzi !

**Comment by mrshirts** [comment]
So, go ahead and convert now?

**Comment by davidlmobley** [comment]
@mrshirts - nope, we will grab you when we're ready. @andrrizzi still needs to generate the files for the REST of the reference calculations he is running. This covers only the SAMPLing component, which is just three guests.

**Comment by mrshirts** [comment]
Just checking in on this.

**Comment by andrrizzi** [comment]
I've started working on this on Monday. I should have the files ready soon. I'll open a new PR and ping you there.
